ipaserver/install/httpinstance.py

```diff
@@ -402,11 +402,14 @@ def __setup_ssl(self):
             if prev_helper is not None:
                 self.add_cert_to_service()
 
+            with open(paths.HTTPD_KEY_FILE, 'rb') as f:
+                priv_key = x509.load_pem_private_key(
+                    f.read(), None, backend=x509.default_backend())
+
             # Verify we have a valid server cert
-            # FIXME: come up with openssl equivalent
-            #  server_certs = db.find_server_certs()
-            #  if not server_certs:
-            #      raise RuntimeError("Could not find a suitable server cert.")
+            if (priv_key.public_key().public_numbers()
+                    != self.cert.public_key().public_numbers()):
+                raise RuntimeError("Could not find a suitable server cert.")
```

Comment:
I'd prefer including a more specific error message like "private key does not match public key." IIRC the previous invocation would also verify the cert trust chain was correct but that may not be as easy to do with openssl which is why had the original FIXME. Comparing the keys would be a good first step and there are other restrictions in place elsewhere so this code may be sufficient.

Reply:
Fixed the error message.

I did not find it anywhere in this path but if there was a cert trust chain verification, it must have happened in certmonger while it was adding the certificate to the NSS DB and thus I am assuming it would still be there.

The original `db.find_server_certs()` only returned a list of certificates the database kept private keys of.